packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -540,7 +570,7 @@ export const MenuBarMixin = (superClass) =>
 
     /** @protected */
     _initButtonAttrs(button) {
-      button.setAttribute('role', 'menuitem');
+      button.setAttribute('role', this.tabNavigation ? 'button' : 'menuitem');
```

Comment:
This line is not tested: when reverting the change, or even removing it, all the unit tests still pass.
I would suggest to add a test that checks the following case with re-setting `items` property:

```js
it('should update role attribute on menu bar buttons when changing items', async () => {
  menu.items = [...menu.items, { text: 'New item' }];
  await nextRender(menu);
  menu._buttons.forEach((btn) => {
    expect(btn.getAttribute('role')).to.equal('menuitem');
  });
});
```

Reply:
```

    it('should update role attribute on menu bar buttons when changing items in tabNavigation', async () => {
      menu.tabNavigation = true;
      await nextRender(menu);
      menu.items = [...menu.items, { text: 'New item' }];
      await nextRender(menu);
      menu._buttons.forEach((btn) => {
        expect(btn.getAttribute('role')).to.equal('button');
      });
      menu.tabNavigation = false;
      await nextRender(menu);
      menu._buttons.forEach((btn) => {
        expect(btn.getAttribute('role')).to.equal('menuitem');
      });
    });

```

Test added